### PR TITLE
fix: contributor fee mitigations

### PR DESCRIPTION
**Pull request by arr00**

### CODE REVIEW

File: contracts/crowdfund/ContributionRouter.sol [hunk 1]
```diff
@@ -76,7 +76,7 @@ contract ContributionRouter {
             assembly {
                 // 228 is the offset of the length of `tokenIds` in the
```
Comment (0xble) [inline contracts/crowdfund/ContributionRouter.sol:79]:
```suggestion
                // 196 is the offset of the length of `tokenIds` in the
```

File: contracts/crowdfund/Crowdfund.sol [hunk 2]
```diff
@@ -366,24 +367,27 @@ abstract contract Crowdfund is Implementation, ERC721Receiver, CrowdfundNFT {
     /// @param initialDelegates The addresses to delegate to for each recipient.
     /// @param values The ETH to contribute for each recipient.
     /// @param gateDatas Data to pass to the gatekeeper to prove eligibility.
-    /// @param revertOnFailure If true, revert if any contribution fails.
     function batchContributeFor(
         address[] memory recipients,
         address[] memory initialDelegates,
-        uint256[] memory values,
-        bytes[] memory gateDatas,
-        bool revertOnFailure
```
Comment (0xble) [inline contracts/crowdfund/Crowdfund.sol:375]: I don't think it does, but note to self that we'll have to check with FE this doesn't break anything.

**Review by 0xble — APPROVED** [review]
Some notes, otherwise looks good. It appears to address each of the following:

- [x]  Fix excess `ETH` is lost when `maxTotalContributions` issue
    - We’re just going to revert when excess ETH is returned (in `receiver()`)
- [x]  Fix `batchContributeFor` calling itself
- [x]  Fix `gateKeeper` checks relying on `msg.sender`

And looks like we are ignoring the following suggestions, which is fine:

- Add timelock for owner changing `feePerMint`
- Allow user making contribution to himself through `ContributionRouter` to set delegate

File: contracts/crowdfund/Crowdfund.sol (hunk 2 above)
Comment (arr00) [reply contracts/crowdfund/Crowdfund.sol:375]: good callout. Agree that it's probably fine.

**Pull request merged**